kratos/utilities/sparse_matrix_multiplication_utility.h

```diff
@@ -120,11 +118,10 @@ class SparseMatrixMultiplicationUtility
         CMatrix& rC
         )
     {
-    #ifdef _OPENMP
-        const int nt = omp_get_max_threads();
-    #else
-        const int nt = 1;
-    #endif
+        int nt = 1;
+        if constexpr(UsesOpenMP) {
```

Comment:
there is a function in the parallel_utils doing exactly this. The bad thing of your approach is that nt is not constant

Reply:
https://github.com/KratosMultiphysics/Kratos/blob/a988fb6b70451e6d55604212a3098223dc947065/kratos/utilities/parallel_utilities.cpp#L34

Reply:
Okay